Mathlib/CategoryTheory/Category/Cat/Terminal.lean

```diff
@@ -0,0 +1,73 @@
+/-
+Copyright (c) 2025 Emily Riehl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robin Carlier, Emily Riehl
+-/
+import Mathlib.CategoryTheory.Category.Cat.Limit
+import Mathlib.CategoryTheory.Limits.Shapes.Terminal
+
+/-!
+# Terminal categories
+
+We prove that a category is terminal if its underlying type has a `Unique` structure and the
+category has an `IsDiscrete` instance.
+
+We then use this to provide various examples of terminal categories.
+
+TODO: Show the converse: that terminal categories have a unique object and are discrete.
+
+TODO: Provide an analogous characterization of terminal categories as codiscrete categories
+with a unique object.
+
+-/
+
+universe v u v' u'
+
+open CategoryTheory Limits Functor
+
+namespace CategoryTheory
+
+namespace Cat
+
+section
+variable {T : Type u} [Category.{v} T] [Unique T] [IsDiscrete T]
+
+/-- The unique functor to the discrete category on a unique object. -/
+def toDiscreteUnique (X : Type u') [Category.{v'} X] : X ⥤ T where
+  obj := fun _ ↦ default
+  map := fun _ ↦ 𝟙 _
```

Comment:
Can you use
```suggestion
/-- The unique functor to the discrete category on a unique object. -/
def toDiscreteUnique (X : Type u') [Category.{v'} X] : X ⥤ T :=
  (const X).obj default
```
and possibly weaken the assumptions on `T` to just be an `[Inhabited T]` category? (dropping `Unique` and `IsDiscrete`)